daemon/sdp.c

```diff
@@ -444,6 +444,12 @@ static int parse_attribute_crypto(struct sdp_attribute *output) {
 	str_shift(&c->key_base64_str, 7);
 	ret = g_base64_decode_step(c->key_base64_str.s, enc_salt_key_len,
 			(guchar *) c->key_salt_buf, &b64_state, &b64_save);
+        // flush b64_state needed for AES-192: 36+2; AES-256: 45+1;
+        if (enc_salt_key_len % 4) {
+                ret += g_base64_decode_step("==", 4 - b64_state,
```

Comment:
I would prefer not to use `b64_state` directly since the glib docs suggest it to be an opaque value and the `g_base64_decode_step` function may actually put a negative number there. It should be possible to use `enc_salt_key_len` to determine how much padding is needed (e.g. something like `4 - (enc_salt_key_len % 4)`)